### PR TITLE
[#2857] Limit message TTL to tenant's max-ttl with Kafka.

### LINKED ISSUE #2857: TTL in downtream messages with Kafka is not capped by max-TTL
The tenant API provides the possibility to define a _max-ttl_ for a tenant. The Kafka-based downstream sender does not limit the TTL by this value.

### PR DESCRIPTION
**Pull request by b-abel**
Fixes #2857.

### CODE REVIEW

File: clients/telemetry-kafka/src/main/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSender.java [hunk 1]
```diff
@@ -154,14 +156,34 @@ public AbstractKafkaBasedDownstreamSender(
...
+                    ttlSeconds = ttl;
+                }
             } else {
-                log.error("TTL property is not a number [{}]", v);
-                return v;
+                if (maxTtl != TenantConstants.UNLIMITED_TTL) {
+                    log.debug("setting TTL to tenant's max TTL [{}s]", maxTtl);
+                    ttlSeconds = maxTtl;
+                } else {
+                    log.error("TTL property is not a number [{}]", v);
```
Comment (sophokles73) [inline clients/telemetry-kafka/src/main/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSender.java:182]: this is also true in the case above, isn't it?

File: clients/telemetry-kafka/src/test/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSenderTest.java [hunk 2]
```diff
@@ -294,44 +294,153 @@ private void assertContentType(final String expectedContentType, final String co
...
+     * configured, the latter one is used.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testThatTtlFromDefaultsIsUsed(final VertxTestContext ctx) {
+        final MockProducer<String, Buffer> mockProducer = KafkaClientUnitTestHelper.newMockProducer(true);
+        final AbstractKafkaBasedDownstreamSender sender = newSender(mockProducer);
+
+        // GIVEN a tenant with maximum TTL configured and a lower one in the properties
```
Comment (sophokles73) [inline clients/telemetry-kafka/src/test/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSenderTest.java:344]: .. a lower one configured in the defaults

File: clients/telemetry-kafka/src/main/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSender.java (hunk 1 above)
Comment (b-abel) [reply clients/telemetry-kafka/src/main/java/org/eclipse/hono/client/telemetry/kafka/AbstractKafkaBasedDownstreamSender.java]: Yes. It was a mistake to add this log statement. The value `v` is especially not a number if it is `null`.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**